src/Microsoft.Health.Dicom.Core/Features/Store/IIndexDataStore.cs

```diff
@@ -92,5 +92,21 @@ public interface IIndexDataStore
         /// <param name="cancellationToken">The cancellation token.</param>
         /// <returns>A task that represents the asynchronous update operation</returns>
         Task<int> IncrementDeletedInstanceRetryAsync(VersionedInstanceIdentifier versionedInstanceIdentifier, DateTimeOffset cleanupAfter, CancellationToken cancellationToken = default);
+
+
+        /// <summary>
+        /// Retrieves the number of deleted instances which have reached the max number of retries.
+        /// </summary>
+        /// <param name="maxNumberOfRetries">The max number of retries.</param>
+        /// <param name="cancellationToken">The cancellation token.</param>
+        /// <returns>A task that gets the count</returns>
+        Task<int> RetrieveNumDeletedMaxRetryCountAsync(int maxNumberOfRetries, CancellationToken cancellationToken = default);
```

Comment:
nit: perhaps the name is more like `RetrieveExhaustedDeletedInstanceAttemptsAsync`?